core/templates/pages/collection-player-page/collection-player-page.directive.spec.ts

```diff
@@ -0,0 +1,535 @@
+    $scope.$apply();
+    $httpBackend.flush();
+    tick();
+
+    expect(alertsSpy).toHaveBeenCalledWith(
+      'There was an error while fetching the collection summary.');
+  }));
+
+  it('should scroll to the given location when calling ' +
+    '\scrollToLocation\'', function() {
+    ctrl.scrollToLocation('location');
+    expect($anchorScroll).toHaveBeenCalled();
+  });
+
+  it('should stop event propagation when click event is emitted', () => {
+    let eventSpy = jasmine.createSpyObj(
+      'event', ['stopPropagation']);
+    ctrl.onClickStopPropagation(eventSpy);
+
+    expect(eventSpy.stopPropagation).toHaveBeenCalled();
+  });
+
+  it('should close exploration card on clicking outside', () => {
+    ctrl.explorationCardIsShown = true;
+    ctrl.closeOnClickingOutside();
+
+    expect(ctrl.explorationCardIsShown).toBe(false);
+  });
+
+  it('should return exploration title position given index ' +
+    'when calling \'getExplorationTitlePosition\'', function() {
+    // Case 1.
+    let result = ctrl.getExplorationTitlePosition(2);
+    expect(result).toBe('8px');
+
+    // Case 2.
+    result = ctrl.getExplorationTitlePosition(1);
+    expect(result).toBe('30px');
+
+    // Case 3.
+    result = ctrl.getExplorationTitlePosition(3);
+    expect(result).toBe('-40px');
+  });
+
+  it('should return exploration url given exploration id', function() {
+    ctrl.collectionId = 'colId1';
+    let url = ctrl.getExplorationUrl('id1');
+
+    expect(url).toBe('/explore/id1?collection_id=colId1');
+  });
+
+  it('should generate path icon parameters', fakeAsync(function() {
+    spyOn(readOnlyCollectionBackendApiService, 'loadCollectionAsync')
+      .and.resolveTo(sampleCollection);
+    // Loading collections.
+    ctrl.$onInit();
+    tick();
+    let pathIconParameters = ctrl.generatePathIconParameters();
+
+    expect(pathIconParameters).toEqual(collectionNodesList);
+  }));
+
+  it('should check whether the exploration is completed', fakeAsync(function() {
+    spyOn(readOnlyCollectionBackendApiService, 'loadCollectionAsync')
+      .and.resolveTo(sampleCollection);
+    // Loading collections.
+    ctrl.$onInit();
+    tick();
+    let result = ctrl.isCompletedExploration();
+
+    expect(result).toEqual(false);
+  }));
+
+  it('should trigger \'$watch\' when collection is ' +
+    'updated', fakeAsync(function() {
+    let $watchSpy = spyOn($scope, '$watch').and.callThrough();
+    sampleCollectionBackendObject.nodes = [collectionNodeBackendObject];
+    sampleCollection = Collection.create(
+      sampleCollectionBackendObject);
+    spyOn(readOnlyCollectionBackendApiService, 'loadCollectionAsync')
+      .and.resolveTo(sampleCollection);
+
+    // Loading collections.
+    ctrl.$onInit();
+    tick();
+
+    ctrl.collection = sampleCollection;
+    $scope.$digest();
+    expect($watchSpy).toHaveBeenCalled();
+  }));
+
+  it('should generate path parameters when collection ' +
+    'node count is one', fakeAsync(function() {
```

Comment:
```suggestion
  it('should not generate path parameters when collection ' +
    'node count is one', fakeAsync(function() {
```
In the test `pathSvgParameters` is set to an empty string. So I'm assuming that path parameters are **not** generated. Am I right?

Reply:
done